config/schemas/config.yaml

```diff
@@ -144,13 +148,18 @@ $defs:
     minimum: 0
     title: Percentage
     type: number
+  port:
+    maximum: 65535
+    minimum: 1
+    type:
+      - integer
+      - string
+    regex: ^[0-9]{0,5}$
```

Comment:
What is the reason for allowing this to be a string?

Reply:
Leniency towards existing configs, see https://github.com/elastisys/compliantkubernetes-apps/pull/2065#discussion_r1570122860 but we should probably push everyone towards integers, but now or later?

Reply:
I would prefer now, but we could do it later.

Reply:
I've ran into issues when not storing numbers as strings, e.g. due to needing leading zero "0777" where the zero was discarded when it was interpreted as a number. I know this is not the case for ports as it has a well defined range but I'm wondering if we want to be consistent in always having numbers as strings or as many numbers as possible as integers and hoping we don't run into issues?

Reply:
We should have number as numbers in the furthest extent possible. Because that is the most intuitive, and JSON schema and YAML should have no problem handling octals, but if it does become a problem we set it as a string for the key that needs it.

I do not see why we should enforce a blanket "numbers as strings" for this since it make the configuration much less intuitive and unnecessary complex. Also we'd loose the ability to validate the numbers.

Reply:
Yeah I guess we will realize when there are issues and just put those numbers as strings. The reason why numbers as strings was suggested is because that's what Helm recommends. But I agree that we'd loose ability to validate numbers that way.

Reply:
So it would be sensible to keep things as integers/numbers in the config and coercing / formatting them into the expected formats at the helmfile → helm boundary then?

I note that `0777` is treated as the integer 777 by [YAML](https://yaml.org/spec/1.2.2/#tag-resolution), it would have to be written `0o777` and then formatted to produce the string `"0777"` if that is the needed format.

Reply:
I see, then some tools do unstandard things with those.

Reply:
So, now or later?

```diff
@@ -151,13 +151,7 @@ $defs:
   port:
     maximum: 65535
     minimum: 1
-    type:
-      - integer
-      - string
-    regex: ^[0-9]{0,5}$
-    $comment: |-
-      The string form should be phased out and replaced by integers which are then
-      coerced into strings during templating.
+    type: integer
   portlist:
     title: Port number
     description: A 16 bit unsigned integer
```

Reply:
I'd rather do it now.

Reply:
Let's do it now then!